### PR TITLE
Add 'Generate constructor with named arguments'

### PR DESCRIPTION
**Pull request by Xylez01**
Added a "Generate" option to generate a constructor with named arguments.

Example:

https://user-images.githubusercontent.com/418977/136557355-6c35322e-c0ae-4456-bd78-ee627bd662e8.mov

Could not find any tests for the actions. If there are any, please point me in the right direction.

### CODE REVIEW

File: Dart/src/com/jetbrains/lang/dart/ide/generation/CreateConstructorWithNamedArgumentsFix.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        template.addTextSegment(",");
+      }
+    }
+    template.addTextSegment(",});");
+    template.addEndVariable();
+    template.addTextSegment(" "); // trailing space is removed when auto-reformatting, but it helps to enter line break if needed
+    return template;
+  }
+
+  private static Boolean isNullable(DartComponent component) {
```
Comment (JordyLangen) [inline Dart/src/com/jetbrains/lang/dart/ide/generation/CreateConstructorWithNamedArgumentsFix.java:68]: Did not find a way to check if a field is nullable, thus implemented this very crude "isNullable" method. Any feedback on this is welcome.

**Comment by asashour** [comment]
FYI, there is a request to do something similar from the server-side, https://github.com/dart-lang/sdk/issues/47389.

However, it is not planned to find all the usage of the constructor(s) and add a default value based on the user input. I hope this limitation can be handled from the IDE side.

**Comment by alexander-doroshko** [comment]
@JordyLangen Thank you for your contribution.

Most of the smart code insight features in Dart are powered by the Analysis Server, a tool from the Dart SDK. IDE only does what the server asks it to do. This includes code highlighting, completion, quick fixes, intention actions, refactorings.

For historical reasons, some features still remain implemented right in the Dart plugin. Code generation actions are among them. Dart plugin knows Dart code worse than the Analysis Server, so these features work not as great as they should.

We are planning to move code generation features to Analysis Server as well. I hope we'll start this work in the nearest months. So I don't think we should work on new code generation features in the Dart plugin.

**Comment by asashour** [comment]
I am not part of the team, but some CLs were accepted, have a look at the `CONTRIBUTING` page.

**Pull request closed without merging**